files/nrpe/check_openstack.py

```diff
@@ -931,21 +937,50 @@ class OSCinderAvailability(cinder.Client):
   options = dict()
 
   def __init__(self, options):
-    self.cinder = cinderclient.client.Client('2', session=keystone_session_v3(options))
+    self.cinder = cinderclient.client.Client('3', session=keystone_session_v3(options))
 
   def get_cinder_volumes(self):
     search_opts = { }
     vols = self.cinder.volumes.list(search_opts=search_opts)
+    print(vols)
     if LOCAL_DEBUG:
-      print vols
+      print(vols)
 
   def execute(self):
-    results = dict()
     try:
       self.get_cinder_volumes()
     except:
       raise
 
+class OSCinderServiceAvailability(cinder.Client):
+
+  def __init__(self, options):
+    self.cinder = cinderclient.client.Client('3', session=keystone_session_v3(options))
+
+  def check_cinder_services(self):
+    msgs = []    
+    services = self.cinder.services.list()
+    
+    for service in services:
+      if service.status == 'enabled' and service.state == 'down':
+        msgs.append("%s on %s, " % (service.binary,service.host))
+    if msgs:
+      raise CinderServiceDownException(msgs=msgs)
+
+    for service in services:
+      if service.status == 'disabled':
+        msgs.append("%s on %s, " % (service.binary,service.host))
+    if msgs:
+      raise CinderServiceDisabledException(msgs=msgs)
+
+    logging.info('No enabled cinder service is down')
```

Comment:
During a platform upgrade we might need to delete the cinder services. E.g. need to delete the Queens version of cinder-volume before the database schema changes for Rocky are permitted.

I was thinking might we want to do some processing for this state. But came to the conclusion this is such a rare case that it's not needed.

Reply:
If you explain further, I can handle it now while I have all the context.

Reply:
This is fine as is, was more just thinking out loud